transformer.py

```diff
@@ -52,6 +57,98 @@
 object_store = None
 
 
+def initialize_logging(request=None):
+    """
+    Get a logger and initialize it so that it outputs the correct format
+
+    :param request: Request id to insert into log messages
+    :return: logger with correct formatting that outputs to console
+    """
+
+    log = logging.getLogger()
+    if 'INSTANCE_NAME' in os.environ:
+        instance = os.environ['INSTANCE_NAME']
+    else:
+        instance = 'Unknown'
+    formatter = logging.Formatter('%(levelname)s ' +
+                                  "{} xaod_cpp_transformer {} ".format(instance, request) +
+                                  '%(message)s')
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    handler.setLevel(logging.INFO)
+    log.addHandler(handler)
+    log.setLevel(logging.INFO)
+    return log
+
+
+def parse_output_logs(logfile):
+    """
+    Parse output from runner.sh and output appropriate log messages
+    :param logfile: path to logfile
+    :return:  None
+    """
+    total_events = 0
+    events_processed = 0
+    total_events_re = re.compile(r'Processing events \d+-(\d+)')
+    events_processed_re = re.compile(r'Processed (\d+) events')
+    with open(logfile, 'r') as f:
+        buf = f.read()
+        match = total_events_re.search(buf)
+        if match:
+            total_events = int(match.group(1))
+        matches = events_processed_re.finditer(buf)
+        for m in matches:
+            events_processed = int(m.group(1))
+        logger.info("{} events processed out of {} total events".format(events_processed, total_events))
+
+
+# class TimeTuple(NamedTuple):
+class TimeTuple(namedtuple("TimeTupleInit", ["user", "system", "idle"])):
+    """
+    Named tuple to store process time information.
+    Immutable so values can't be accidentally altered after creation
+    """
+    # user: float
+    # system: float
+    # idle: float
```

Comment:
Suggest a [dataclass](https://docs.python.org/3/library/dataclasses.html) here, since they're more declarative, easy to make immutable, and you also get a nice `__repr__`.
```suggestion
@dataclass(frozen=True)
class TimeTuple:
    """
    Dataclass to store process time information.
    Immutable so values can't be accidentally altered after creation.
    """
    user: float
    system: float
    idle: float
```

Reply:
dataclass was added in python 3.7, this code runs in python 2.7.  I would have gone with a different solution but my hands are tied.